src/components/ObsDetails/FlagItemModal.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+
+import CheckBox from "@react-native-community/checkbox";
+import createFlag from "api/flags";
+import Button from "components/SharedComponents/Buttons/Button";
+import {
+  Modal,
+  SafeAreaView,
+  Text, View
+} from "components/styledComponents";
+import { t } from "i18next";
+import type { Node } from "react";
+import React, { useRef, useState } from "react";
+import {
+  findNodeHandle
+} from "react-native";
+import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
+import { TextInput } from "react-native-paper";
+import IconMaterial from "react-native-vector-icons/MaterialIcons";
+import useAuthenticatedMutation from "sharedHooks/useAuthenticatedMutation";
+
+type Props = {
+  id:number,
+  showFlagItemModal: boolean,
+  closeFlagItemModal: Function
+}
+
+const FlagItemModal = ( {
+  id, showFlagItemModal, closeFlagItemModal
+}: Props ): Node => {
+  const keyboardScrollRef = useRef( null );
+  const [value, setValue] = useState( "none" );
```

Comment:
What is `value` referring to here? Is this the type of flag or something else? Can we use more descriptive language with this and `toggleValue`?